revamped grouplist structure in courses and learnergroups, restyled grouplists

thecoolestguy opened the pull request:
revamped grouplist structure in courses and learnergroups, restyled grouplists, refs #64

jrjohnson commented:
In general I think @thecoolestguy is correct, this entire component should be called resultlist.  or results-list.  Specific comments in line messages.

jrjohnson commented on app/styles/components/_grouplist.scss:
```diff
@@ -1,21 +1,25 @@
+      overflow: auto;
+      padding: .2em;
+      .grouplist-title {
+        float: left;
```
Instead of floating these left and right they should be placed on the grid.  See https://github.com/ilios/frontend/blob/master/app/styles/components/_detailview.scss#L46 for an example.  Basically the grouplist already spans 12 of 12 so the title should maybe be 3 of 12 and the h2 9 of 12.

jrjohnson commented on app/templates/learnergroups.hbs:
```diff
@@ -34,15 +34,20 @@
-  <section class='grouplist'>
-    <h2>{{title}}</h2>
+
+  <section class='grouplist clearfix'>
```
Instead of using the section class clearfix use the sass @include clearfix; on the element.

saschaben commented:
looks good. I think that the word "results" is probably not what we want in the final iteration, but will do for now as a placeholder. we should get user feedback on nomenclature for this one.

Merging.

saschaben commented:
or rather --- merge as soon as ready!

thecoolestguy commented:
I renamed the grouplist div's classes to 'resultslist' and removed the floats from the div's in favor of using Neat, but I'm still trying to get a hang of using Neat, so it may not be correct.

Also, I noticed this morning that there was a border around the filter tools that I'd never seen before, so I removed that.

jrjohnson commented:
Code changes look good.  Nice work.  Needs a rebase to merge.
